pandas/core/index.py

```diff
@@ -973,7 +976,9 @@ def _convert_list_indexer(self, keyarr, kind=None):
         and we have a mixed index (e.g. number/labels). figure out
         the indexer. return None if we can't help
         """
-        if (kind is None or kind in ['iloc','ix']) and (is_integer_dtype(keyarr) and not self.is_floating()):
+        if (kind is None or kind in ['iloc', 'ix']) and (
+            is_integer_dtype(keyarr) and not self.is_floating() and not isinstance(keyarr, ABCPeriodIndex)):
```

Comment:
Agree for most cases - when you have a `PeriodIndex` and you supply another as an index. 
But if you have an integer index and someone supplies a `PeriodIndex` as the key, I don't think you want that resolving to its `int`s. Does that make sense?

Would this be solved if we had a `Period` `dtype`? I couldn't find an open issue about that, but I think you've mentioned it a few times